### PR TITLE
Adding telemetry data for PTR task

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
Adding telemetry data for the task.
`
##[debug]Processed: ##vso[telemetry.publish area=TestExecution;feature=PublishTestResultsTask]{"builduri":"vstfs:///Build/Build/564","buildid":"564","osType":"Windows_NT","testRunner":"JUnit","failTaskOnFailedTests":"true","mergeResultsUserPreference":"false","config":null,"platform":null,"testResultsFilesCount":2,"failedTestsInRun":true,"subFeature":"publishTestResultsTaskConsolidatedCiEvent"}`

### CODE REVIEW

File: Tasks/PublishTestResultsV2/cieventlogger.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ import * as tl from 'vsts-task-lib/task';
 
 const area: string = 'TestExecution';
```
Comment (ShreyasRmsft) [inline Tasks/PublishTestResultsV2/cieventlogger.ts]: change area

Comment (ShreyasRmsft) [reply Tasks/PublishTestResultsV2/cieventlogger.ts]: to TestReporting or TestManagement whatever their area is

File: Tasks/PublishTestResultsV2/publishtestresults.ts [hunk 2]
```diff
@@ -66,6 +67,8 @@ async function run() {
         tl.debug('publishRunAttachments: ' + publishRunAttachments);
         tl.debug('failTaskOnFailedTests: ' + failTaskOnFailedTests);
 
+        
+
```
Comment (ShreyasRmsft) [inline Tasks/PublishTestResultsV2/publishtestresults.ts]: nit: remove extra newlines

File: Tasks/PublishTestResultsV2/publishtestresultstool.ts [hunk 3]
```diff
@@ -67,9 +67,7 @@ export class TestResultsPublisher {
             fs.writeFileSync(responseFilePath, fileContent);
         } catch (ex) {
             // Log telemetry and return null path
-            publishEvent({
-                "exception": ex
-            });
+            publishEvent('publishTestResultsTaskEvent',{"exception": ex});
```
Comment (ShreyasRmsft) [inline Tasks/PublishTestResultsV2/publishtestresultstool.ts]: add to consolidated ci itself

Comment (ShreyasRmsft) [reply Tasks/PublishTestResultsV2/publishtestresultstool.ts]: nit: fix formatting

File: Tasks/PublishTestResultsV2/cieventlogger.ts [hunk 4]
```diff
@@ -13,8 +14,17 @@ function getDefaultProps() {
     };
 }
 
-export function publishEvent(properties: { [key: string]: any }): void {
+export function addToConsolidatedCi(key: string, value: any) {
+    consolidatedCiData[key] = value;
+}
+
+export function fireConsolidatedCi() {
```
Comment (ShreyasRmsft) [inline Tasks/PublishTestResultsV2/cieventlogger.ts]: add a call to this in a finally block around the main code

File: Tasks/PublishTestResultsV2/cieventlogger.ts (hunk 1 above)
Comment (vagisha-nidhi) [reply Tasks/PublishTestResultsV2/cieventlogger.ts]: Area is already `TestExecution`

File: Tasks/PublishTestResultsV2/publishtestresults.ts (hunk 2 above)
Comment (vagisha-nidhi) [reply Tasks/PublishTestResultsV2/publishtestresults.ts]: Done

File: Tasks/PublishTestResultsV2/publishtestresultstool.ts (hunk 3 above)
Comment (vagisha-nidhi) [reply Tasks/PublishTestResultsV2/publishtestresultstool.ts]: Done

File: Tasks/PublishTestResultsV2/cieventlogger.ts (hunk 4 above)
Comment (vagisha-nidhi) [reply Tasks/PublishTestResultsV2/cieventlogger.ts]: Done

File: Tasks/PublishTestResultsV2/publishtestresultstool.ts [hunk 5]
```diff
@@ -67,9 +68,9 @@ export class TestResultsPublisher {
             fs.writeFileSync(responseFilePath, fileContent);
         } catch (ex) {
             // Log telemetry and return null path
-            publishEvent({
-                "exception": ex
-            });
+            ci.addToConsolidatedCi('exception', ex);
+            ci.fireConsolidatedCi();
```
Comment (ShreyasRmsft) [inline Tasks/PublishTestResultsV2/publishtestresultstool.ts]: move to finally block

Comment (vagisha-nidhi) [reply Tasks/PublishTestResultsV2/publishtestresultstool.ts]: Done as discussed

**Pull request merged**